aws-sdk-core/lib/aws-sdk-core.rb

```diff
@@ -268,6 +270,12 @@ module Xml
 
   class << self
 
+    # @api private
+    def shared_config
+      enabled = ENV["AWS_SDK_LOAD_CONFIG"] ? true : false
+      @shared_config ||= SharedConfig.new(config_enabled: true)
+    end
+
```

Comment:
Minor nitpick. I try to limit the top-level utility functions. A simple change would be to keep this function inside the `Aws::SharedConfig` module. You could name the method something like `.default_config`, e.g `Aws::SharedConfig.default_config`.